### ISSUE TITLE
Support .mkosi directory

### ISSUE THREAD

**Issue by lucaswerkmeister** [systemd/mkosi#176]
We now have 5 `mkosi.*` files and directories that people might want to commit to source code repositories plus 6 (7 with #165) more that they may want to have locally. I think that’s enough that it would be nice to support dropping them all in a hidden `.mkosi` directory instead of in the root directory.

To avoid confusion, we should probably only support “everything’s in `.`” and “everything’s in `.mkosi`”, not any mixed modes where some files are in `.` and some are in `.mkosi`.

**Comment by poettering** [comment]
Fix waiting in #165

**Comment by keszybz** [comment]
I don't think we should use a hidden directory. Those contents are huge after all. Apart from the name, #165 answers this RFE.

**Comment by lucaswerkmeister** [comment]
Unless I misunderstand #165, I don’t see what it has to do with this issue… the RFE is to have only a single `*mkosi*` entry in the repository root. With #165, AFAIU (though I haven’t tried it out yet), `mkosi.default`, `mkosi.build` etc. are all still outside of `mkosi.output/`.

**Comment by poettering** [comment]
Ah, right, I figure I should actually read the issue properly, before working on them... Reopening.

**Comment by lucaswerkmeister** [comment]
I suppose GitHub’s `.github/` (where you can drop `README.md`, `CONTRIBUTING.md`, and I don’t remember what else) is a good analogy to this.

**Comment by keszybz** [comment]
I misunderstood too. For configuration files, `.mkosi/` sounds good.

I guess we could check `.mkosi/mkosi.default` if `mkosi.default` is not found, and `.mkosi/mkosi.build` if `mkosi.build` is not found, etc.

**Comment by poettering** [comment]
hmm, i'd really like an "mkosi -a" option, i.e. a single command to build "all" flavours of your image, if you want to build multiple. In the systemd tree we already have a .mkosi/ dir with files for fedora, arch, debian. I think it would make a ton of sense if "mkosi -a" would mean "build everything dropped into .mkosi/"

**Comment by keszybz** [comment]
Maybe it makes sense to add a different prefix or suffix for such distro configuration files. There's already `mkosi.build` and `mkosi.extra`, `.postinst`, and a bunch of others, so I see the point of moving that all to a subdir. So many files can be a lot of clutter in the top dir.

**Comment by poettering** [comment]
PR #309 implements this, though the dir is called "mkosi.files/"
